### ISSUE TITLE
☂️  Cleanup `SemanticsFlag` and `SemanticsAction`. 

### ISSUE THREAD

**Issue by bernaferrari** [flutter/flutter#123346]
~These two classes should be an enum, because they work as enum, but they are older than the enhanced enum, so this wasn't possible before. Now it is.~

We can remove 200 LOC of these classes and make them as similar as they can be to enums. Enum has its own index, so it clashes with the index from them and would be a super breaking change because affects Google Testing. But changing everything else and cleanup isn't. 

This is kind of breaking because Flutter uses semanticsFlag.values.values, but I searched on code search and couldn't find a repo that uses that (besides code-gen), so it seems really internal.

- [x] part 1: [engine] add `valuesAsList` (https://github.com/flutter/engine/pull/40571)
- [x] part 2: [flutter] use `valuesAsList` (https://github.com/flutter/flutter/pull/123817)
- [x] part 3: [engine] ~convert to enum~ cleanup (https://github.com/flutter/engine/pull/40567)
- [x] part 4: [flutter] use `values` again (https://github.com/flutter/flutter/pull/123329)
- [x] part 5: [engine] remove `valuesAsList` (https://github.com/flutter/engine/pull/41126)
- [x] part 6: [flutter] deprecate `describeEnum` (https://github.com/flutter/flutter/pull/125016)
